homeassistant/components/shelly/button.py

```diff
@@ -112,12 +113,10 @@ def async_migrate_unique_ids(
     if not entity_entry.entity_id.startswith("button"):
         return None
 
-    device_name = slugify(coordinator.device.name)
-
     for key in ("reboot", "self_test", "mute", "unmute"):
-        old_unique_id = f"{device_name}_{key}"
+        old_unique_id = f"{coordinator.mac}_{key}"
```

Comment:
This migration is over two years old now, so I think we can use it for a new purpose.